server.go

```diff
@@ -308,7 +309,12 @@ func NewFromConfig(logger *logrus.Logger, conf Config) (*Server, error) {
 	}
 
 	if conf.SignalfxAPIKey != "" {
-		sfxSink, err := signalfx.NewSignalFxSink(conf.SignalfxAPIKey, conf.SignalfxEndpointBase, conf.SignalfxHostnameTag, conf.Hostname, ret.TagsAsMap, ret.Statsd, log, nil)
+		fallback := signalfx.NewClient(conf.SignalfxEndpointBase, conf.SignalfxAPIKey)
```

Comment:
This section feels like it should be in the sink, since it's leaking the sink's import in here, no?

Reply:
Good point - I think the way this is done here is just about right (allows you to stub that client in tests), but having to import the dpsink package in server.go sucks. I'm making another type in the `signalfx` sink package that proxies that type forward.